src/MICore/NativeMethods.cs

```diff
@@ -8,15 +8,15 @@ namespace MICore
 {
     internal class NativeMethods
     {
-        // TODO: It would be better to route these to the correct .so files directly rather than pasing through system.native. 
+        private const string Libc = "libc";
```

Comment:
Maybe I am missing something, but I don't see an explicit check that we are on Linux. It seems like we may need some additional calls to RuntimeInformation.IsOSPlatform(OSPlatform.Linux).
Also, it seems like we might want to rename this class to 'UnixNativeMethods' or maybe 'LinuxNativeMethods' if this code doesn't work on OSX either.